middleware/kubernetes/README.md

```diff
@@ -1,204 +1,264 @@
 # kubernetes
 
-*kubernetes* enables reading zone data from a kubernetes cluster.
-It implements the [spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md)
-defined for kubernetes DNS-Based service discovery:
+The *kubernetes* middleware enables the reading zone data from a Kubernetes cluster.  It implements the [Kubernetes DNS-Based Service Discovery Specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md). 
 
-Service `A` records are constructed as "myservice.mynamespace.svc.coredns.local" where:
+CoreDNS running the kubernetes middleware can be used as a replacement of kube-dns in a kubernetes cluster.  See the [deployment](https://github.com/coredns/deployment) repository for details on [how to deploy CoreDNS in Kubernetes](https://github.com/coredns/deployment/tree/master/kubernetes).
 
-* "myservice" is the name of the k8s service
-* "mynamespace" is the k8s namespace for the service, and
-* "svc" indicates this is a service
-* "coredns.local" is the zone
+## Syntax
+
+```
+kubernetes ZONE [ZONE...] [{
+	[resyncperiod PERIOD]
+	[endpoint URL
+	[tls CERT-FILE KEY-FILE CACERT-FILE]]
+	[namespaces NAMESPACE [NAMESPACE...]]
+	[labels EXPRESSION [,EXPRESSION...]]
+	[pods POD-MODE]
+	[cidrs CIDR [CIDR...]]
+	[upstream ADDRESS [ADDRESS...]]
+	[federation NAME DOMAIN]
+	[autopath [NDOTS [RESPONSE [RESOLV-CONF]]]
+	[fallthrough]
+}]
+```
 
-Pod `A` records are constructed as "1-2-3-4.mynamespace.pod.coredns.local" where:
+* **resyncperiod PERIOD**
+	
+  The Kubernetes data API resynchronization period. Default is 5m. Example values: 60s, 5m, 1h
 
-* "1-2-3-4" is derived from the ip address of the pod (1.2.3.4 in this example)
-* "mynamespace" is the k8s namespace for the service, and
-* "pod" indicates this is a pod
-* "coredns.local" is the zone
+  Example:
 
-Endpoint `A` records are constructed as "epname.myservice.mynamespace.svc.coredns.local" where:
+  ```
+	kubernetes cluster.local. {
+		resyncperiod 15m
+	}
+  ```
 
-* "epname" is the hostname (or name constructed from IP) of the endpoint
-* "myservice" is the name of the k8s service that the endpoint serves
-* "mynamespace" is the k8s namespace for the service, and
-* "svc" indicates this is a service
-* "coredns.local" is the zone
+* **endpoint URL**
```

Comment:
~~~
`endpoint`
~~~
for the name
~~~
**URL**
~~~
for the variable your referencing, see middleware.md